### PR TITLE
Update prometheus to v2.40.1

### PR DESCRIPTION
**Pull request by fpetkovski**
This commit updates the prometheus/prometheus module to v2.40.1

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,13 +5,13 @@ go 1.17
...
 	github.com/go-openapi/runtime v0.24.0
-	github.com/go-openapi/strfmt v0.21.2
+	github.com/go-openapi/strfmt v0.21.3
 	github.com/metalmatze/signal v0.0.0-20210307161603-1c9aa721a97a
 	github.com/oklog/run v1.1.0
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/alertmanager v0.24.0
-	github.com/prometheus/client_golang v1.12.1
-	github.com/prometheus/prometheus v0.35.0
+	github.com/prometheus/client_golang v1.13.1
```
Comment (PhilipGough) [inline go.mod:13]: @fpetkovski I wonder if we should take the opportunity to bump this to `1.14.0`?

Comment (fpetkovski) [reply go.mod]: Good point, updated.

**Review by PhilipGough — APPROVED** [review]
lgtm

**Review by simonpasquier — APPROVED** [review]
thanks!

**Pull request merged**